spec/system/candidate_interface/candidate_new_user_with_course_params_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe 'A new candidate arriving from Find with a course and provider code' do
+  include FindAPIHelper
+  include ActiveJob::TestHelper
+
+  scenario 'retaining their course selection through the sign up process' do
+    given_the_pilot_is_open
+    and_confirm_course_choice_from_find_is_activated
+    and_the_course_i_selected_only_has_one_site
+
+    when_i_arrive_from_find_to_a_course_that_is_open_on_apply
+    and_i_click_apply_on_apply
+    then_the_url_should_contain_the_course_code_and_provider_code_param
+    when_i_fill_in_the_eligiblity_form_with_yes
+    then_the_url_should_contain_the_course_code_and_provider_code_param
+
+    when_i_submit_my_email_address
+    and_click_on_the_magic_link
+    then_i_should_see_the_courses_review_page
+    and_i_should_see_the_course_name_and_code
+    and_i_should_see_the_site(@site)
+    and_my_find_from_id_course_should_be_set_to_nil
+
+    given_the_course_i_selected_has_multiple_sites
+
+    when_i_submit_my_email_address
+    and_click_on_the_magic_link
+    then_i_should_see_the_course_choices_site_page
+    and_i_should_see_the_site(@site1)
+    and_i_should_see_the_site(@site2)
```

Comment:
What would you think about combining these into a step like `and_i_see_the_form_to_pick_a_location`?